style/properties/shorthands/text.mako.rs

```diff
@@ -56,64 +56,73 @@
-        #[allow(unused)]
         fn to_css<W>(&self, dest: &mut CssWriter<W>) -> fmt::Result where W: fmt::Write {
             use crate::values::specified::TextDecorationLine;
-
-            let (is_solid_style, is_current_color, is_auto_thickness) =
-            (
+            use crate::values::specified::Color;
+    
             % if engine == "gecko":
-                *self.text_decoration_style == text_decoration_style::SpecifiedValue::Solid,
-                *self.text_decoration_color == specified::Color::CurrentColor,
-                self.text_decoration_thickness.is_auto()
+                let (is_solid_style, is_current_color, is_auto_thickness) =
+                (
+                    *self.text_decoration_style == text_decoration_style::SpecifiedValue::Solid,
+                    *self.text_decoration_color == Color::CurrentColor,
+                    self.text_decoration_thickness.is_auto()  
+                );
             % else:
-                true, true, true
+                let (is_solid_style, is_current_color) =
+                (
+                    *self.text_decoration_style == text_decoration_style::SpecifiedValue::Solid,
+                    *self.text_decoration_color == Color::CurrentColor
+                );
             % endif
-            );
-
+    
             let mut has_value = false;
             let is_none = *self.text_decoration_line == TextDecorationLine::none();
-            if (is_solid_style && is_current_color && is_auto_thickness) || !is_none {
-                self.text_decoration_line.to_css(dest)?;
-                has_value = true;
-            }
 
-            % if engine == "gecko":
-            if !is_auto_thickness {
-                if has_value {
-                    dest.write_char(' ')?;
-                }
-                self.text_decoration_thickness.to_css(dest)?;
+            if (is_solid_style && is_current_color 
+                % if engine == "gecko":
+                    && is_auto_thickness
+                % endif
+            ) || !is_none {
+                self.text_decoration_line.to_css(dest)?;
                 has_value = true;
             }
-
+    
             if !is_solid_style {
                 if has_value {
                     dest.write_char(' ')?;
                 }
                 self.text_decoration_style.to_css(dest)?;
                 has_value = true;
             }
-
+    
             if !is_current_color {
                 if has_value {
                     dest.write_char(' ')?;
                 }
                 self.text_decoration_color.to_css(dest)?;
                 has_value = true;
             }
+    
+            % if engine == "gecko":
+                if !is_auto_thickness {
+                    if has_value {
+                        dest.write_char(' ')?;
+                    }
+                    self.text_decoration_thickness.to_css(dest)?;
+                    has_value = true;
+                }
```

Comment:
Don't move this to the end.

Reply:
Done. Moved it to it's previous position.